kolibri/content/api.py

```diff
@@ -0,0 +1,274 @@
+    """
+    Set prerequisite relationship between content1 and content2.
+
+    :param channel_id: str
+    :param content1: ContentMetadata or str
+    :param content2: ContentMetadata or str
+    """
+    KolibriContent.PrerequisiteContentRelationship.objects.using(channel_id).create(
+        relationship_type='prerequisite', contentmetadata_1=content1, contentmetadata_2=content2)
+
+@can_get_content_with_id
+def set_is_related(channel_id=None, content1=None, content2=None, **kwargs):
+    """
+    Set is related relationship between content1 and content2.
+
+    :param channel_id: str
+    :param content1: ContentMetadata or str
+    :param content2: ContentMetadata or str
+    """
+    KolibriContent.RelatedContentRelationship.objects.using(channel_id).create(
+        relationship_type='related', contentmetadata_1=content1, contentmetadata_2=content2)
+
+@can_get_content_with_id
+def children_of_kind(channel_id=None, content=None, kind=None, **kwargs):
+    """
+    Get all ContentMetadatas of a particular kind under the given ContentMetadata.
+    For kind argument, please pass in a string like "topic" or "video" or "exercise".
+
+    :param channel_id: str
+    :param content: ContentMetadata or str
+    :param kind: str
+    :return: QuerySet of ContentMetadata
+    """
+    return content.get_descendants(include_self=False).filter(kind=kind).using(channel_id)
+
+def update_content_copy(file_object=None, content_copy=None):
+    """
+    Update the File object you pass in with the content copy
+    You can pass None on content_copy to remove the associated file on disk.
+
+    :param file_object: File
+    :param content_copy: str
+    """
+    if not file_object:
+        raise TypeError("must provide a File object to update content copy")
+    if content_copy:
+        file_object.content_copy = DjFile(open(content_copy, 'rb'))
+    else:
+        file_object.content_copy = None
+
+    file_object.save()
+
+
+"""channel API methods"""
+
+def get_channel(channel_identifier):
+    """
+    Get a ChannelMetadata object by channel id or channel name.
+
+    :param channel_identifier: str
+    :return: ChannelMetadata
+    """
+    if validate.is_valid_uuid(channel_identifier):
+        try:
+            the_channel = KolibriContent.ChannelMetadata.objects.get(channel_id=channel_identifier)
+        except KolibriContent.ChannelMetadata.DoesNotExist:
+            raise ValueError("cannot find any channel with this channel id")
+        return the_channel
+    elif isinstance(channel_identifier, str):
+        try:
+            the_channel = KolibriContent.ChannelMetadata.objects.get(name=channel_identifier)
+        except KolibriContent.ChannelMetadata.DoesNotExist:
+            raise ValueError("cannot find any channel with this channel name")
+        except KolibriContent.ChannelMetadata.MultipleObjectsReturned:
+            raise ValueError("more than one channel named " + "'" + channel_identifier + "'")
```

Comment:
I am not sure this error is helpful. In production, it would simply break a system with no recourse.